lib/p2p/P2P.js

```diff
@@ -24,6 +24,19 @@ class P2P extends EventEmitter {
     });
   }
 
+  /**
+   * Construct peer's id
+   * @param {string} host
+   * @param {string} address
+   * @param {number} port
+   * @returns {string}
+   */
+  static getPeerId(host, address, port) {
+    return !host || address === host
+      ? `${address}:${port}`
+      : `${host}[${address}]:${port}`;
```

Comment:
Why use the host name here? Would it not be the case here that if I tried to connect to the same remote node twice, once by host name and once by IP address, I'd get different peerIds each time? I'm also wondering if we should call this something other than "peerId" perhaps? I'm thinking that name conflicts with database columns. Maybe call this something like `peerString`?

Reply:
That all makes sense. I agree we don't need to keep the host name past the point of logging the initial connection. I made the keys for the `peers` object use ip address and port since that is essentially the unique identifying tuple for a peer, going with just ip address doesn't work for when multiple nodes are running on a single address which is the case with local testing.

Logging the host name in the "connected to" message would be nice, see my newest comment for how we could do that without having to put `host` in the peerId/peerString.

I think with that change and renaming `getPeerId` to `getPeerString`, as well as removing the host from the peer string, this should be good to merge. I'm looking forward to your wider scope followup PR on this.

Reply:
That solution was not feasible because there were more dependencies on that string (the one with the host), and what you're called `peerString` is basically a key (although it's a string), so a different solution was required.